wordpress-mobile/WordPress-Android
# My Site Dashboard [Phase 2]: Navigate to Edit Post [HACK - DO NOT MERGE]

Parent: #15709

This PR implements this functionality that will allow the user to navigate to the `Edit Post` screen, from the `My Site` screen, through the `Posts` screen.

PS: The PR is marked as `HACK` because of the two `hack` related commits that are trying to bend the paging framework to allow such navigation. @ashiagr and @zwarm I am adding you both as reviewers since I want both your feedback on this PR, its solution and more specifically I want to to focus on the two `hack` related commits. Maybe you can recommend another way of doing that.

-----

Prerequisite:

- Go to `My Site` -> `Me` -> `App Settings` -> `Debug Settings` configuration.
- Turn-on the `MySiteDashboardPhase2FeatureConfig` feature.
- Relaunch the app.

-----

To test:
- If you don't have a draft or scheduled post already, create one.
- Note that on the `My Site` tab, your draft or scheduled posts, if any, are shown.
- Click on a draft or scheduled posts and verify that:
  - First, the `Posts` screen is shown for a second or two.
  - Then, the `Edit Post` screen is launched with the content of the draft or scheduled post that was selected.

The above is testing the happy path. In addition to that please make sure to test any or all other scenarios, ultimately trying to break the solution. For example:
- Before clicking on the draft or scheduled post, update it remotely. Then, verify that by clicking on it, when navigated to the `Edit Post` screen, then you will get the remote updates, instead of the now stale/out-dated local state of the post.
- Enter the `Offline` mode and try to test the solution. You might need to comment out the `CardsSource.postState(...)` for when an error occurs during `cardsStore.fetchCards(selectedSite)` (see line 83). Otherwise, when entering the `Offline` mode, the dashboard feed will disappear. PS: This scenario will be also handling via the ` Handle offline mode` subtask, but it would be great to test it now anyway.

NOTE: It is expected that the `Posts` screen is shown. A subsequent PR, which will handle the full screen post loading, will make sure that the loading experience will be seamless and that the user will never notice the intermediate `Posts` screen.

-----

## Regression Notes
1. Potential unintended areas of impact

Data Loss (⚠️): Due to the nature of the solution, the user can enter into a situation where by clicking on a draft or scheduled post, the `Edit Post` screen will open a stale version of the post, or an unwanted version. At which point, if the user clicks back on a draft post, publishes this draft post, or rescheduled/publishes a scheduled post, then they might experience unintentional data loss.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Extensive manual testing. See `To test` sections above.

3. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:
